raft/raft.hh

```diff
@@ -300,6 +300,15 @@ struct transport_error: public error {
     using error::error;
 };
 
+// Can be thrown by two-way RPCs when the destination is not seen as alive by the failure detector.
+// If thrown, the request was not sent at all.
+struct destination_not_alive_error: public transport_error {
+    server_id destination;
+
+    destination_not_alive_error(std::source_location sloc, server_id destination)
```

Comment:
We should be using `seastar::compat::source_location`, `std::source_location` is broken IIRC

Reply:
Done, replaced with `seastar::compat::source_location`